fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerProfessionBuilder.java

```diff
@@ -41,7 +39,11 @@
  * <p>A corresponding <code>IDENTIFIER_PATH.mcmeta</code> file exits in the same directory to define properties such as the {@link VillagerResourceMetadata.HatType HatType} this profession would use.
  *
  * <p>Note this does not register any trades to these villagers. To register trades, add a new entry with your profession as the key to {@link TradeOffers#PROFESSION_TO_LEVELED_TRADE}.
+ *
+ * @deprecated Replaced by access widener for {@link VillagerProfession#VillagerProfession}
+ * in Fabric Transitive Access Wideners (v1).
  */
+@Deprecated
```

Comment:
IMO we should mark when they will be removed. Is it going to be 1.19 or 1.20?

Reply:
They will only be removed if vanilla makes heavy breaking changes in this area. Otherwise we will keep them "forever".

Reply:
Sure love the community project absolutely wanting to be more stable than the game it mods. /s

It becomes ridiculous.

Reply:
This PR isn't honestly the place to talk about this - we probably have old deprecated APIs from 1.14 and 1.15 still lying around so this isn't much different.

Reply:
Given mojang nukes stuff often, like text refactors or registry tags, I don't mind nuking more often as long has a clear replacement is offered.

Reply:
I had recent experiences with AE2 where Forge changed existing APIs unnecessarily, and I'm pretty sure MI is still using deprecated APIs because I haven't updated it to the new ones yet. I think Player had similar experiences. So, we keep APIs as long as maintaining them is reasonable.

If you disagree, then feel free to do it differently in your projects, but this is not going to change in Fabric anytime soon.

Reply:
We probably should start marking the version for deprecation and allow removing them eagerly in contributions if the version differences to the current version is large enough.

Reply:
> allow removing them eagerly in contributions if the version differences to the current version is large enough

that will be done by the maintainers already, I don't think something needs to be done about it. :P